### PR TITLE
Create

### PR DESCRIPTION
**Pull request by Casanova569**
Summary:
Add the function of create a movie to the application.
Objetive: The trello card is considered approved if a movie can be created.

Trello Card:
Name: Creado
Link: https://trello.com/c/8T2W21tu/8-creado

### CODE REVIEW

**Comment by mikolaslorant** [comment]
Pull request titles should be in english :)

File: src/Views/MovieApp/Create.cshtml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                <label asp-for="Price" class="control-label"></label>
+                <input asp-for="Price" class="form-control" />
+                <span asp-validation-for="Price" class="text-danger"></span>
+            </div>
+            <div class="form-group">
+                <input type="submit" value="Save" class="btn btn-primary" />
+            </div>
+        </form>
+    </div>
+</div>
```
Comment (mikolaslorant) [inline src/Views/MovieApp/Create.cshtml]: add eof

File: src/Repositories/DataBase/MvcMovieContext.cs [hunk 2]
```diff
@@ -4,12 +4,12 @@
...
 {
     public class MvcMovieContext : DbContext
     {
-        public MvcMovieContext (DbContextOptions<MvcMovieContext> options){}
+        public MvcMovieContext (DbContextOptions<MvcMovieContext> options): base(options){}
 
-        public DbSet<MvcMovie.Models.Movie> Movies { get; set; }
+        public DbSet<Movie> Movies { get; set; }
     }
 }
```
Comment (mikolaslorant) [inline src/Repositories/DataBase/MvcMovieContext.cs]: Add EOF

File: src/Repositories/DataBase/MvcMovieContext.cs [hunk 3]
```diff
@@ -4,12 +4,12 @@
 using System.Threading.Tasks;
 using Microsoft.EntityFrameworkCore;
 
-namespace MvcMovie.Repositories.DataBase
+namespace MvcMovie.Models
 {
     public class MvcMovieContext : DbContext
     {
-        public MvcMovieContext (DbContextOptions<MvcMovieContext> options){}
+        public MvcMovieContext (DbContextOptions<MvcMovieContext> options): base(options){}
```
Comment (mikolaslorant) [inline src/Repositories/DataBase/MvcMovieContext.cs]: add space after base(options)

**Review by mikolaslorant — CHANGES_REQUESTED** [review]

File: src/Controllers/MovieApp.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+using Microsoft.AspNetCore.Mvc;
+using Queries.Core;
+using MvcMovie.Models;
+
+namespace MvcMovie.Controllers
+{
+    public class MovieAppController : Controller
```
Comment (truccomarcos) [inline src/Controllers/MovieApp.cs]: Class and file names should match. Update the file name to match the class name

File: src/Controllers/MovieApp.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        private readonly IUnitOfWork _unitOfWork;
+
+        public MovieAppController(IUnitOfWork unitOfWork)
+        {
+            this._unitOfWork = unitOfWork;
+        }
+        public IUnitOfWork UnitOfWork
+        {
+            get { return this._unitOfWork; }
+        }
```
Comment (truccomarcos) [inline src/Controllers/MovieApp.cs]: Add a blank line here to separate the index method from the property. Also, add the [HttpGet] tag to it

File: src/Controllers/MovieApp.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+        [HttpPost]
+        public IActionResult Create(Movie movie)
+        {
+            UnitOfWork.Movies.Add(movie);
+            UnitOfWork.Complete();
+            return View();
+        }
+    }
+}
```
Comment (truccomarcos) [inline src/Controllers/MovieApp.cs]: Add empty line at EOF

File: src/Controllers/MovieApp.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+namespace MvcMovie.Controllers
+{
+    public class MovieAppController : Controller
+    {
+        private readonly IUnitOfWork _unitOfWork;
+
+        public MovieAppController(IUnitOfWork unitOfWork)
+        {
+            this._unitOfWork = unitOfWork;
+        }
```
Comment (truccomarcos) [inline src/Controllers/MovieApp.cs]: Add empty line

File: src/Controllers/MovieApp.cs (hunk 5 above)
Comment (truccomarcos) [reply src/Controllers/MovieApp.cs]: We usually put properties like this one below the Endpoints (in this case, after the POST Create)

File: src/Controllers/MovieApp.cs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public IActionResult Create(Movie movie)
+        {
+            UnitOfWork.Movies.Add(movie);
+            UnitOfWork.Complete();
+            return View();
```
Comment (truccomarcos) [inline src/Controllers/MovieApp.cs]: Try redirecting the user to the index page for this controller (even though the list is not ready yet, the view is there

**Review by truccomarcos — CHANGES_REQUESTED** [review]

File: src/Controllers/MovieAppController.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+        [HttpGet]
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public IActionResult Create([Bind("ID,Title,ReleaseDate,Genre,Price")] Movie movie)
```
Comment (mikolaslorant) [inline src/Controllers/MovieAppController.cs]: As I pointed out int other prs. Don't use bind and make the method receive a MovieViewModel. This allows for independence between the view and the database model. Also It's  a good practice to remove  The binding is resolved automatically by the framework.

File: src/Controllers/MovieAppController.cs [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            UnitOfWork.Movies.Add(movie);
```
Comment (mikolaslorant) [inline src/Controllers/MovieAppController.cs]: Also before adding the movie you could add a check of the model to see if its valid

**Review by mikolaslorant — CHANGES_REQUESTED** [review]

**Review by truccomarcos — APPROVED** [review]

**Pull request merged**